### DISCUSSION TITLE
environment variables are undefined in production

### DISCUSSION THREAD

**Discussion by sheheryar-pirzada** [chimurai/http-proxy-middleware#801] [Q&A]
I'm trying to set the target key in HPM through an env variable which I have in `.env.production`.
However, I get this error:
`throw new Error(errors_1.ERRORS.ERR_CONFIG_FACTORY_TARGET_MISSING);`
`Error: [HPM] Missing "target" option. Example: {target: "http://www.example.org"}`

The env variable is undefined for some reason.
Anyone know what the issue might be?

`app.use('/api', createProxyMiddleware({ target: process.env.REACT_APP_API_URL, changeOrigin: true }));`

**Comment by chimurai** [comment]
Your question is more related to `dotenv`. (https://www.npmjs.com/package/dotenv)

Following their documentation should answer your question.

**Comment by moweiwei** [comment]
I meet the same question in nextjs.  Do you konw how to resolve it?
